### ISSUE TITLE
WString.h is not compatible to c++11 range-based loop

### ISSUE THREAD

**Issue by Rotzbua** [arduino/Arduino#4837]
I tried to use the new c++11 syntax with the default string class from arduino. Actually it fails because it has no `begin` or `end`.

Test-IDE: 1.6.8 (Win8.1 x64)
Board: Uno

Example code:
```arduino
String temp = "abc";
for (auto tempchar : temp) {
    Serial.println(tempchar);
}
```

Error:
```
error: 'begin' was not declared in this scope
for (auto tempchar : temp) {
error: 'end' was not declared in this scope
```
Reference:
http://en.cppreference.com/w/cpp/language/range-for

Best

**Comment by NicoHood** [comment]
This was already "discussed" somewhere else. No answer as I remember. I also suggest to add this feature.

**Comment by Chris--A** [comment]
I have a PR hanging around: https://github.com/arduino/Arduino/pull/2179

I proposed this before C++11 was enabled.

**Comment by cmaglie** [comment]
Fixed with #2179
